### PR TITLE
Upgrade to pyo3 0.3.0 

### PR DESCRIPTION
**Pull request by konstin**
This combines some code improvements which an upgrade to pyo3 0.3.0. Since 0.3.0 is to be released (most likely in a few day; [#188](https://github.com/PyO3/pyo3/issues/188)), it is for now pinned to the latest commit on pyo3's master.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -120,7 +113,7 @@ fn init(py: Python, m: &PyModule) -> PyResult<()> {
         )
     }
 
-    m.add("__version__", version!());
+    m.add("__version__", env!("CARGO_PKG_VERSION"));
```
Comment (mre) [inline src/lib.rs]: simply beautiful

File: src/lib.rs [hunk 2]
```diff
@@ -85,8 +78,8 @@ impl From<pyo3::PyErr> for HyperJsonError {
 import_exception!(json, JSONDecodeError);
 
 /// A hyper-fast JSON encoder/decoder written in Rust
-#[py::modinit(_hyperjson)]
-fn init(py: Python, m: &PyModule) -> PyResult<()> {
+#[pymodinit]
+fn hyperjson(py: Python, m: &PyModule) -> PyResult<()> {
```
Comment (mre) [inline src/lib.rs]: This is also a nice one.

Comment (konstin) [reply src/lib.rs]: Yep, I've just implemented that in pyo3 inspired by #27.

**Comment by konstin** [comment]
Should be ready to merge now :)

**Comment by mre** [comment]
Thanks @konstin. Much cleaner. 🌻

**Pull request merged**